var/spack/repos/builtin/packages/lapackpp/package.py

```diff
@@ -45,16 +45,27 @@ class Lapackpp(CMakePackage):
 
     def cmake_args(self):
         spec = self.spec
-        return [
+
+        args = [
             '-DBUILD_SHARED_LIBS=%s' % ('+shared' in spec),
             '-Dbuild_tests=%s'       % self.run_tests,
             '-DLAPACK_LIBRARIES=%s'  % spec['lapack'].libs.joined(';')
         ]
 
+        if spec['blas'].name == 'cray-libsci':
+            args.append(self.define('BLA_VENDOR', 'CRAY'))
+
+        return args
+
     def check(self):
         # If the tester fails to build, ensure that the check() fails.
         if os.path.isfile(join_path(self.build_directory, 'test', 'tester')):
             with working_dir(self.build_directory):
                 make('check')
         else:
             raise Exception('The tester was not built!')
+
+    def flag_handler(self, name, flags):
+        if (spec['blas'].name == 'cray-libsci') and name == 'cxxflags':
```

Comment:
I get an error: `Error: NameError: name 'spec' is not defined`.

Perhaps use `self.spec` instead.

Reply:
Thanks @teonnik , I made your suggested change.